### PR TITLE
feat: add option to set partition expiration on tables

### PR DESCRIPTION
**Pull request by ingarabr**

### CODE REVIEW

File: core/src/test/scala/no/nrk/bigquery/internal/TableUpdateOperationTest.scala [hunk 1]
```diff
@@ -437,4 +438,53 @@ class TableUpdateOperationTest extends FunSuite {
...
+                .setField("date")
+                .setExpirationMs(expiration.map(exp => Long.box(exp.toMillis)).orNull)
+                .build()
+            )
+            .build()
+        )
+        .build()
+    )
+
+    TableUpdateOperation.from(testTable(Some(1.day)), remote(None)) match {
```
Comment (HenningKoller) [inline core/src/test/scala/no/nrk/bigquery/internal/TableUpdateOperationTest.scala:470]: Maybe add one more test case where we remove expiration from remote?

**Review by HenningKoller — APPROVED** [review]
👍

File: build.sbt [hunk 2]
```diff
@@ -109,7 +109,12 @@ lazy val core = crossProject(JVMPlatform)
     Compile / doc / scalacOptions ++= Seq(
       "-no-link-warnings" // Suppresses problems with Scaladoc @throws links
     ),
-    mimaBinaryIssueFilters := Nil
+    mimaBinaryIssueFilters := List(
```
Comment (hamnis) [inline build.sbt:112]: when bumping series above this is not needed.

File: core/src/test/scala/no/nrk/bigquery/internal/TableUpdateOperationTest.scala (hunk 1 above)
Comment (ingarabr) [reply core/src/test/scala/no/nrk/bigquery/internal/TableUpdateOperationTest.scala:470]: Good point. I amended it to the original commit.

File: build.sbt (hunk 2 above)
Comment (ingarabr) [reply build.sbt]: Ah, makes sense. I've removed them.

**Review by hamnis — APPROVED** [review]

**Pull request merged**